spark/src/test/java/org/apache/iceberg/spark/source/TestPartitionPruning.java

```diff
@@ -0,0 +1,417 @@
+      return buckets.contains(bucketId);
+    };
+
+    runTest(filterCond, partCondition);
+  }
+
+  @Test
+  public void testPartitionPruningTruncatedString() {
+    String filterCond = "message like 'info event%'";
+    Predicate<Row> partCondition = (Row r) -> {
+      String truncatedMessage = r.getString(3);
+      return truncatedMessage.equals("info ");
+    };
+
+    runTest(filterCond, partCondition);
+  }
+
+  @Test
+  public void testPartitionPruningTruncatedStringComparingValueShorterThanPartitionValue() {
+    String filterCond = "message like 'inf%'";
+    Predicate<Row> partCondition = (Row r) -> {
+      String truncatedMessage = r.getString(3);
+      return truncatedMessage.startsWith("inf");
+    };
+
+    runTest(filterCond, partCondition);
+  }
+
+  @Test
+  public void testPartitionPruningHourlyPartition() {
+    String filterCond;
+    if (spark.version().startsWith("2")) {
+      // Looks like from Spark 2 we need to compare timestamp with timestamp to push down the filter.
+      filterCond = "timestamp >= to_timestamp('2020-02-03T01:00:00')";
+    } else {
+      filterCond = "timestamp >= '2020-02-03T01:00:00'";
+    }
+    Predicate<Row> partCondition = (Row r) -> {
+      int hourValue = r.getInt(4);
+      Instant instant = getInstant("2020-02-03T01:00:00");
+      Integer hourValueToFilter = hourTransform.apply(TimeUnit.MILLISECONDS.toMicros(instant.toEpochMilli()));
+      return hourValue >= hourValueToFilter;
+    };
+
+    runTest(filterCond, partCondition);
+  }
+
+  private void runTest(String filterCond, Predicate<Row> partCondition) {
+    File originTableLocation = createTempDir();
+    Assert.assertTrue("Temp folder should exist", originTableLocation.exists());
+
+    Table table = createTable(originTableLocation);
+    Dataset<Row> logs = createTestDataset();
+    saveTestDatasetToTable(logs, table);
+
+    List<Row> expected = logs
+        .select("id", "date", "level", "message", "timestamp")
+        .filter(filterCond)
+        .orderBy("id")
+        .collectAsList();
+    Assert.assertFalse("Expected rows should be not empty", expected.isEmpty());
+
+    // remove records which may be recorded during storing to table
+    CountOpenLocalFileSystem.resetRecordsInPathPrefix(originTableLocation.getAbsolutePath());
+
+    List<Row> actual = spark.read()
+        .format("iceberg")
+        .option("vectorization-enabled", String.valueOf(vectorized))
+        .load(table.location())
+        .select("id", "date", "level", "message", "timestamp")
+        .filter(filterCond)
+        .orderBy("id")
+        .collectAsList();
+    Assert.assertFalse("Actual rows should not be empty", actual.isEmpty());
+
+    Assert.assertEquals("Rows should match", expected, actual);
+
+    assertAccessOnDataFiles(originTableLocation, table, partCondition);
+  }
+
+  private File createTempDir() {
+    try {
+      int rand = random.nextInt(1000000);
+      return temp.newFolder(String.format("logs-%d", rand));
```

Comment:
Why does this create its own random number instead of just getting a new temp folder? I don't see much benefit to doing it this way.

Reply:
I didn't realize `temp.newFolder()` would just work for creating unique dir. My bad.